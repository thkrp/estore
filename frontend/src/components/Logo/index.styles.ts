import styled from 'styled-components';
import Link from '../Links/Link';

export const WrapperStyled = styled.div`
    flex: 0 0 30%;
    min-width: 250px;
`;

export const LogoStyled = styled(Link)`
    display: block;
    width: 100%;
    height: 100%;
    img {
        width: 100%;
    }
`;
