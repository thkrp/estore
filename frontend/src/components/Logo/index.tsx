import React, { FC } from 'react';
import { LogoStyled, WrapperStyled } from './index.styles';
import { imageUrl } from '../../common/helpers/path.helper';

type Props = {
    logo?: string;
};
const Logo: FC<Props> = ({ logo }) => {
    return (
        <WrapperStyled>
            <LogoStyled to="/">
                <img src={imageUrl(logo)} alt="logo" />
            </LogoStyled>
        </WrapperStyled>
    );
};

export default Logo;
