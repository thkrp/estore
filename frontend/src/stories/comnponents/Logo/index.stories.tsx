import { Meta, StoryObj } from '@storybook/react';
import { withRouter } from 'storybook-addon-react-router-v6';
// eslint-disable-next-line no-restricted-imports
import LogoComponent from '../../../components/Logo';

const meta = {
    title: 'Components/Logo',
    component: LogoComponent,
    decorators: [withRouter],
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs'],
    argTypes: {
        logo: {
            description: 'logo path'
        }
    }
} satisfies Meta<typeof LogoComponent>;

export default meta;
type Story = StoryObj<typeof meta>;
export const Logo: Story = {
    args: {
        logo: ''
    }
};
