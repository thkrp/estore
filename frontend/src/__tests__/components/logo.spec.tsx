/* eslint-disable no-restricted-imports*/
import React from 'react';
import '@testing-library/jest-dom';
import { cleanup, render, screen } from '@testing-library/react';
import { mockData } from 'app-shared';
import { MemoryRouter } from 'react-router-dom';
import * as sanitizer from '@braintree/sanitize-url';
import Logo from '../../components/Logo';

describe('logo', () => {
    afterEach(cleanup);

    it('should be able to pass a logo', () => {
        const { container } = render(<Logo logo={mockData.generalInformation.logo} />, { wrapper: MemoryRouter });
        const logo = screen.getByRole('img');
        expect(logo).toHaveAttribute('src', '/upload/iblock/1a8/1a8a5771cc80a6d2ef5a93e271ebfd1f.png');
        expect(container).toMatchSnapshot();
    });

    it('sanitizeUrl has been called', () => {
        jest.spyOn(sanitizer, 'sanitizeUrl');
        render(<Logo logo={mockData.generalInformation.logo} />, { wrapper: MemoryRouter });
        expect(sanitizer.sanitizeUrl).toHaveBeenCalled();
    });

    it('should contain alt text', () => {
        render(<Logo logo={mockData.generalInformation.logo} />, { wrapper: MemoryRouter });
        const logo = screen.getByRole('img');
        expect(logo).toHaveAttribute('alt', 'logo');
    });

    it('should return default logo', () => {
        render(<Logo />, { wrapper: MemoryRouter });
        const logo = screen.getByRole('img');
        expect(logo).toHaveAttribute('src', 'no-image.png');
    });
});
