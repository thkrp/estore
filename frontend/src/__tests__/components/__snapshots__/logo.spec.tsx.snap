// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo should be able to pass a logo 1`] = `
<div>
  <div
    class="sc-beySPh gHTqDZ"
  >
    <a
      class="sc-guDLey jlzTwU"
      href="/"
    >
      <img
        alt="logo"
        src="/upload/iblock/1a8/1a8a5771cc80a6d2ef5a93e271ebfd1f.png"
      />
    </a>
  </div>
</div>
`;
